home/olizilla/2021-04-exploration-report-lotus-openrpc.md

```diff
@@ -0,0 +1,306 @@
+- Use GitHub actions to automatically publish updates to the `js-lotus-client`.
+- Upgrade `api.chain.love` to lotus >= 1.17 and ensure it's OpenRPC discovery method `rpc.discover` is publically readable without an api key, and publish links to the OpenRPC playground and inspector for it, to make the OpenRPC more widely known.
+- Contribute fixes to docs and Typescript generation back to https://github.com/open-rpc/generator
+
+## Inspecting the OpenRPC definiton
+
+You can fetch if via the standard `rpc.discver` method
+
+```console
+ curl -H "Content-Type: application/json" \
+  --data '{ "jsonrpc": "2.0", "method": "rpc.discover", "params": [], "id": 1 }' \
+  'http://44.236.109.158:28000/rpc/v0' | jq
+```
+
+<details>
+  <summary>Show response snippet</summary>
+
+```json
+{
+  "jsonrpc": "2.0",
+  "result": {
+    "info": {
+      "title": "Lotus RPC API",
+      "version": "1.6.0-dev"
+    },
+    "methods": [
+      {
+        "deprecated": false,
+        "description": "```go\nfunc (s *FullNodeStruct) BeaconGetEntry(p0 context.Context, p1 abi.ChainEpoch) (*types.BeaconEntry, error) {\n\treturn s.Internal.BeaconGetEntry(p0, p1)\n}\n```",
+        "externalDocs": {
+          "description": "Github remote link",
+          "url": "https://github.com/filecoin-project/lotus/blob/master/api/apistruct/struct.go#L885"
+        },
+        "name": "Filecoin.BeaconGetEntry",
+        "paramStructure": "by-position",
+        "params": [
+          {
+            "deprecated": false,
+            "description": "abi.ChainEpoch",
+            "name": "p1",
+            "required": true,
+            "schema": {
+              "description": "Number is a number",
+              "examples": [
+                10101
+              ],
+              "title": "number",
+              "type": [
+                "number"
+              ]
+            },
+            "summary": ""
+          }
+        ],
+        "result": {
+          "deprecated": false,
+          "description": "*types.BeaconEntry",
+          "name": "*types.BeaconEntry",
+          "required": true,
+          "schema": {
+            "additionalProperties": false,
+            "examples": [
+              {
+                "Data": "Ynl0ZSBhcnJheQ==",
+                "Round": 42
+              }
+            ],
+            "properties": {
+              "Data": {
+                "media": {
+                  "binaryEncoding": "base64"
+                },
+                "type": "string"
+              },
+              "Round": {
+                "title": "number",
+                "type": "number"
+              }
+            },
+            "type": [
+              "object"
+            ]
+          },
+          "summary": ""
+        },
+        "summary": "BeaconGetEntry returns the beacon entry for the given filecoin epoch. If\nthe entry has not yet been produced, the call will block until the entry\nbecomes available\n"
+      },
+  ```
```

Comment:
leading space on this line is messing up formatting

Reply:
Fixed!